semantic-diff/src/Data/Algebra.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE DeriveGeneric, DeriveTraversable, RankNTypes #-}
+module Data.Algebra
```

Comment:
I think eventually we’re going to either remove this or replace it with something better-suited to precise ASTs in `semantic-ast`, but this is a good home for it for the time being 👍